runtime/domain.c

```diff
@@ -821,46 +821,48 @@ static void unreserve_minor_heaps(void) {
   caml_mem_unmap((void *) caml_minor_heaps_start, size);
 }
 
-static void stw_resize_minor_heap_reservation(caml_domain_state* domain,
-                                       void* minor_wsz_data,
-                                       int participating_count,
-                                       caml_domain_state** participating) {
-  barrier_status b;
-  uintnat new_minor_wsz = (uintnat) minor_wsz_data;
+static void domain_resize_heap_reservation(uintnat new_minor_wsz)
```

Comment:
Add a comment pointing out that this is only called by the final domain at the barrier, before the barrier is lifted. Hmm, should we have a function naming convention like `_stw` for this?

Reply:
We should, and this is exactly the topic of #12619, which is stuck waiting for a complete review. Feedback is welcome if you have any. (I haven't looked at the present PR yet, and I expect from eyeballing your comments from a distance that there will be conflicts between the two PRs. Oh well, such is life. Let whichever PR is reviewed and ready first be merged, and the other one will have to be rebased.)